README.md

```diff
@@ -162,9 +181,86 @@ specifies channel 0 which is connected to the thermometer.
+We pattern match with `{:ok, pid}` so we can use `pid` in the 
+SPI.transfer function next.
+
+    {:ok, pid} = SPI.start_link("spidev0.0")
+
+`elixir_ale` uses the Linux spidev device driver for communicating with SPI 
+devices. It creates files in /dev for each SPI interface that it knows about. 
+Many boards only have one SPI interface and it shows up as /dev/spidev0.0. 
+The easiest way to find out the right name is to run `ls spidev*` in the /dev 
+directory. If nothing shows up, it is likely that the spidev driver needs to 
+be loaded. Try running modprobe spidev or consult your board's documentation.
+
+Next we'll make a call to SPI.transfer to read the potentiometer.
+
+    SPI.transfer(pid, <<0x78, 0x00>>)
+
+As mentioned above, the protocol for talking to the ADC is described in the 
+[MCP3002](http://www.microchip.com/wwwproducts/en/MCP3002) data sheet. See 
+Figure 6-1, the request protocol is labelled "MCU Transmitted Data." The 
+leftmost bit is the "Start" bit. The second bit is SGL/DIFF, the third bit is 
+ODD/SIGN, and the fourth bit is MSBF. You may notice a `x` before the start 
+bit which represents a leading `0`.  This is often done when dealing with 
+microcontroller-based SPI ports that must send 8 bits at a time. The request 
+and response are happening simultaneously so the size of the request (in bits) 
+must equal the size of the response. In this case, the request consists of 4 
+bits and the response contains 11 bits (a null bit and 10 data bits) for a 
+total of 15 bits, so a leading `0` is necessary to get 16 bits.
+
+Table 5-1 explains how to configure the request bits. For example, SGL/DIFF==1 
+and ODD/SIGN==1 specifies single-ended mode from channel 1 which has been 
+connected to the potentiometer. MSBF==1 will output the result of the conversion 
+with MSBF (Most Significant Bit First.) [Here](https://en.wikipedia.org/wiki/Most_significant_bit#Most_Significant_Bit_First_vs_Least_Significant_Bit_First) is
+a brief summary of MSBF vs LSBF (Least Significant Bit First) if you are not 
+familiar with the difference. Finally, the ADC doesn't care about the 11 bits 
+at the end of the request corresponding with the response so we can just send 
+11 `0` bits at the end.
+
+Going back to Figure 6-1 with our bit configuration from Table 5-1, the complete 
+request to read the pontentiometer on channel 1 should be `01111000b 00000000b`. This 
+is often abbreviated using the hexidecimal numbers corresponding to the bits, 
+which is in this case `0x78` and `0x00`. Alternatively, sending `0x68` and `0x00` or 
+`01101000b 00000000b` would read the temperature on channel 0.
+
+Now use use binary pattern matching to pull out the ADC counts (low 10 bits).
+
+    <<_::size(6), counts::size(10)>> = 
+
+Here we're pattern matching using the size of bitstrings.  If this is new to 
+you, you might want to refer to the 
+[Elixir docs for bitstrings](https://hexdocs.pm/elixir/Kernel.SpecialForms.html#%3C%3C%3E%3E/1).
```

Comment:
I tested it and it seems to work for me.  I'll add an "or run `h <<>>` at the IEx prompt."